tests/unit_tests/test_tethys_cli/test_proxyapps_commands.py

```diff
@@ -0,0 +1,223 @@
+    @mock.patch("tethys_cli.proxyapps_commands.write_success")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_update_proxy_apps(self, mock_exit, mock_write_success, mock_write_info):
+        mock_args = mock.Mock()
+        mock_args.name = self.app_name
+        mock_args.set_kwargs = [["logo_url", "https://fake.com"]]
+
+        self.assertRaises(
+            SystemExit,
+            update_proxyapp,
+            mock_args,
+        )
+        mock_write_info.assert_called_with(
+            "Attribute logo_url was updated successfully with https://fake.com"
+        )
+        mock_write_success.assert_called_with(
+            f"Proxy app '{self.app_name}' was updated successfully"
+        )
+        mock_exit.assert_called_with(0)
+
+    @mock.patch("tethys_cli.proxyapps_commands.write_error")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_add_proxy_apps_with_existing_proxy_app(self, mock_exit, mock_write_error):
+        mock_args = mock.Mock()
+        mock_args.name = self.app_name
+        mock_args.endpoint = "http://foo.example.com/my-proxy-app"
+
+        self.assertRaises(
+            SystemExit,
+            add_proxyapp,
+            mock_args,
+        )
+        mock_write_error.assert_called_with(
+            f"There is already a proxy app with that name: {self.app_name}"
+        )
+        mock_exit.assert_called_with(1)
+
+    @mock.patch("tethys_cli.proxyapps_commands.write_error")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_add_proxyapp_integrity_error(self, mock_exit, mock_write_error):
+        app_name_mock = "My_Proxy_App_for_Testing_2"
+        mock_args = mock.Mock()
+        mock_args.name = app_name_mock
+        mock_args.endpoint = "http://foo.example.com/my-proxy-app"
+        mock_args.description = None
+        mock_args.logo_url = None
+        mock_args.tags = None
+        mock_args.enabled = None
+        mock_args.show_in_apps_library = None
+        mock_args.back_url = None
+        mock_args.open_new_tab = None
+        mock_args.display_external_icon = None
+        mock_args.order = None
+
+        self.assertRaises(
+            SystemExit,
+            add_proxyapp,
+            mock_args,
+        )
+        mock_write_error.assert_called_with(
+            f"Not possible to add the proxy app: {app_name_mock}"
+        )
+        mock_exit.assert_called_with(1)
+
+    @mock.patch("tethys_cli.proxyapps_commands.write_success")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_add_proxyapp_success(self, mock_exit, mock_write_success):
+        app_name_mock = "My_Proxy_App_for_Testing_2"
+        mock_args = mock.Mock()
+        mock_args.name = app_name_mock
+        mock_args.endpoint = "http://foo.example.com/my-proxy-app"
+        mock_args.description = ""
```

Comment:
Add another test that adds a new proxy app with non-default values for all arguments and verify with a query that the values are all saved in the database correctly.